### PR TITLE
Documents refactoring

### PR DESCRIPTION
**Pull request by TarasOrlovskyi**
1. TTN in two copies
2. Print only 2 Invoices for cash orders (buhBaseProperty=false in InvoiceDto)
3. Add product articles for specific invoices and bills

### CODE REVIEW

File: goods-partner-backend/src/main/java/com/goodspartner/report/CarsLoadReportGenerator.java [hunk 1]
```diff
@@ -74,10 +77,17 @@ private CarLoadDetails buildCarLoadDetails(CarLoad carLoad) {
...
                 .flatMap(order -> order.getProducts().stream()
                         .map(product -> getProductLoadDetails(product, order)))
                 .collect(Collectors.groupingBy(productLoadDetails -> productLoadDetails.getProduct().getProductName()));
+        return getSortedProductLoadDetailsMap(productLoadDetailsMap);
+    }
+
+    private Map<String, List<ProductLoadDetails>> getSortedProductLoadDetailsMap(Map<String, List<ProductLoadDetails>> productLoadDetails) {
+        return productLoadDetails.entrySet().stream()
+                .sorted(Map.Entry.comparingByKey())
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue, (e1, e2) -> e1, LinkedHashMap::new));
```
Comment (belosh59) [inline goods-partner-backend/src/main/java/com/goodspartner/report/CarsLoadReportGenerator.java:90]: why don't do this in above method?

File: goods-partner-backend/src/main/java/com/goodspartner/mapper/decorator/ODataInvoiceMapperDecorator.java [hunk 2]
```diff
@@ -56,23 +57,23 @@ private void setInvoiceSeatsAmount(InvoiceDto invoice, ValueConverters converter
...
+        invoice.setTextNumericPDV(String.format(textNumericPdv, pdvUnits, pdvMoneyDto.getNanos()));
     }
 
     private void setTextInvoiceAmount(InvoiceDto invoice, ValueConverters converters) {
         MoneyDto invoiceAmount = getMoneyDto(invoice.getInvoiceAmount(), 2);
-        String invoiceAmountUnits = converters.asWords(invoiceAmount.getUnits());
-        String invoiceAmountNanos = converters.asWords(invoiceAmount.getNanos());
-        invoice.setTextInvoiceAmount(String.format("%s гривень %s копійок", invoiceAmountUnits, invoiceAmountNanos));
+        String invoiceAmountUnits = StringUtils.capitalize(converters.asWords(invoiceAmount.getUnits()));
+//        String invoiceAmountNanos = converters.asWords(invoiceAmount.getNanos());
```
Comment (belosh59) [inline goods-partner-backend/src/main/java/com/goodspartner/mapper/decorator/ODataInvoiceMapperDecorator.java:75]: why keep it?

File: goods-partner-backend/src/main/java/com/goodspartner/mapper/decorator/ODataInvoiceMapperDecorator.java [hunk 3]
```diff
@@ -56,23 +57,23 @@ private void setInvoiceSeatsAmount(InvoiceDto invoice, ValueConverters converter
...
+//        String priceNanos = converters.asWords(priceMoneyDto.getNanos());
 
         MoneyDto pdvMoneyDto = getMoneyDto(invoice.getInvoiceAmountPDV(), 2);
         String pdvUnits = converters.asWords(pdvMoneyDto.getUnits());
-        String pdvNanos = converters.asWords(pdvMoneyDto.getNanos());
-
-        invoice.setTextNumeric(
-                priceUnits + " гривень " + priceNanos + " копійок \n" +
-                        "У т.ч. ПДВ: " + pdvUnits + " гривень " + pdvNanos + " копійок");
+//        String pdvNanos = converters.asWords(pdvMoneyDto.getNanos());
```
Comment (belosh59) [inline goods-partner-backend/src/main/java/com/goodspartner/mapper/decorator/ODataInvoiceMapperDecorator.java:65]: why keep this comment, we really dont need it anymore?

File: goods-partner-backend/src/main/java/com/goodspartner/service/document/impl/PdfFileCompiler.java [hunk 4]
```diff
@@ -151,67 +152,80 @@ private String getPdfOfBillAndInvoicesAndImagesQualityDocuments(DocumentContent
         return tempPdfFileName;
     }
 
-    private void setFontArial() throws IOException {
-        ITextFontResolver fontResolver = renderer.getFontResolver();
-        fontResolver.addFont(FONT_ARIAL, BaseFont.IDENTITY_H, BaseFont.EMBEDDED);
+    private void insertDocuments(List<String> documentList, FileOutputStream pdfHost) {
```
Comment (belosh59) [inline goods-partner-backend/src/main/java/com/goodspartner/service/document/impl/PdfFileCompiler.java:155]: host?

File: goods-partner-backend/src/main/java/com/goodspartner/service/document/impl/PdfFileCompiler.java [hunk 5]
```diff
@@ -151,67 +152,80 @@ private String getPdfOfBillAndInvoicesAndImagesQualityDocuments(DocumentContent
...
+        for (int i = 0; i < documentList.size(); i++) {
+            if (i == 0) {
+                insertFirstDocument(documentList.get(i), pdfHost);
+            } else {
+                insertNextDoconemt(documentList.get(i));
+            }
+        }
```
Comment (belosh59) [inline goods-partner-backend/src/main/java/com/goodspartner/service/document/impl/PdfFileCompiler.java:162]: I dont understand what happens here

File: goods-partner-backend/src/main/java/com/goodspartner/service/dto/external/grandedolce/ODataInvoiceProductDto.java [hunk 6]
```diff
@@ -50,41 +51,53 @@ public class ODataInvoiceProductDto {
...
     private ProductMeasureDetails productUnit;
     private ProductMeasureDetails productPackaging;
+    private ProductMeasureDetails productReport;
 
     @JsonProperty("Номенклатура")
     private void mapProductName(Map<String, Object> value) {
         this.productName = (String) value.get("Description");
         this.productRefKey = (String) value.get("Ref_Key");
         this.productGTDRefKey = (String) value.get("НоменклатураГТД");
+        this.article = (String) value.get("Артикул");
```
Comment (belosh59) [inline goods-partner-backend/src/main/java/com/goodspartner/service/dto/external/grandedolce/ODataInvoiceProductDto.java:61]: String.valueOf ?

File: goods-partner-backend/src/main/java/com/goodspartner/service/document/impl/PdfFileCompiler.java (hunk 4 above)
Comment (TarasOrlovskyi) [reply goods-partner-backend/src/main/java/com/goodspartner/service/document/impl/PdfFileCompiler.java:155]: it's Sergii imagination

File: goods-partner-backend/src/main/java/com/goodspartner/service/document/impl/PdfFileCompiler.java (hunk 5 above)
Comment (TarasOrlovskyi) [reply goods-partner-backend/src/main/java/com/goodspartner/service/document/impl/PdfFileCompiler.java:162]: It's specific logic for first element, but for other elements another logic

**Review by Rooman — APPROVED** [review]

**Review by Rooman — APPROVED** [review]

File: goods-partner-backend/src/main/java/com/goodspartner/service/dto/external/grandedolce/ODataInvoiceProductDto.java (hunk 6 above)
Comment (TarasOrlovskyi) [reply goods-partner-backend/src/main/java/com/goodspartner/service/dto/external/grandedolce/ODataInvoiceProductDto.java:61]: if value.get("Артикул") is null?

**Pull request merged**